File: 3BIN/BINV3140-1 Programmation - question spéciales/BINV3140-A .NET/Séance 1-20200919/2. Exercices/BDActeurs/BDActeurs/src/be/ipl/film_database/domaine/ListePersonnes.java
package be.ipl.film_database.domaine;

import java.security.InvalidParameterException;
import java.util.HashMap;
import java.util.Iterator;
import java.util.Map;

public class ListePersonnes {

	private static ListePersonnes instance;
	private Map<String, Personne> lesPersonnes;
	
	/**
	 * Constructeur privé car singleton.
	 */
	private ListePersonnes(){
		lesPersonnes = new HashMap<String, Personne>();
	}
	
	/**
	 * ListePersnnes est un singleton.
	 * @return La seule intancede la classe
	 */
	public static ListePersonnes getInstance(){
		
		if (instance == null)
			instance = new ListePersonnes();
		
		return instance;
	}
	
	/**
	 * Ajoute une personne à la liste en utilisant sn nom comme clef
	 * @param personne
	 */
	public void ajouterPersonne(Personne personne){
		if (personne == null)
			throw new InvalidParameterException();
		lesPersonnes.put(personne.getNom(), personne);
	}
	
	public Iterator<Personne> listePersonne(){
		return lesPersonnes.values().iterator();
	}
	
}

File: 3BIN/BINV3140-1 Programmation - question spéciales/BINV3140-A .NET/Séance 1-20200919/2. Exercices/Week_1/Week_1/ListePersonnes.cs
using System;
using System.Collections.Generic;

namespace be.ipl.film_database.domaine
{
    class ListePersonnes
    {
		private static ListePersonnes _instance;
		private IDictionary<String, Personne> _lesPersonnes;

		/**
		 * Constructeur privé car singleton.
		 */
		private ListePersonnes()
		{
			_lesPersonnes = new Dictionary<String, Personne>();
		}

		/**
		 * ListePersnnes est un singleton.
		 * @return La seule intancede la classe
		 */
		public static ListePersonnes GetInstance()
		{

			if (_instance == null)
				_instance = new ListePersonnes();

			return _instance;
		}

		/**
		 * Ajoute une personne à la liste en utilisant sn nom comme clef
		 * @param personne
		 */
		public void AjouterPersonne(Personne personne)
		{
			if (personne == null)
				throw new ArgumentNullException();
			_lesPersonnes.Add(personne.Nom, personne);
		}

		public IEnumerator<Personne> ListePersonne()
		{
			return _lesPersonnes.Values.GetEnumerator();
		}
	}
}
